File: 02-moderate/remove_characters/remove.characters.cpp
// Reuben Thorpe (2016), CodeEval [Remove Characters v1.2]
#include <iostream>
#include <fstream>
#include <sstream>
#include <algorithm>

using namespace std;


int main(int argc, char* argv[]) {
  ifstream in_file(argv[1]);
  string line;

  while (getline(in_file, line)) {
    stringstream sstr(line);
    string sentence, remove_chars;
    getline(sstr, sentence, ',');
    sstr >> remove_chars;

    for (auto n: remove_chars) {
      sentence.erase(remove(sentence.begin(),
                            sentence.end(), n),
                            sentence.end());
    }

    cout << sentence << '\n';
  }

  return(0);
}

File: code.eval/02-moderate/remove_characters/remove.characters.py
# Reuben Thorpe (2016), CodeEval [Remove Characters v1.1]
from sys import argv


def parse_problem(file_path):
    with open(argv[1], 'r') as in_file:
        for line in in_file:
            yield(line.strip().split(", "))


if __name__ == '__main__':
    for problem in parse_problem(argv[1]):
        sentence, remove_char = problem

        for char in remove_char:
            sentence = sentence.replace(char, '')

        print(sentence)
